File: MIDTERM-LAB-3/JavaScript/MP01.js
const fs = require('fs');
const readline = require('readline');
const path = require('path');

function validateFilePath(filePath) {
    if (!fs.existsSync(filePath)) {
        console.log("Error: File does not exist.");
        return false;
    }
    if (path.extname(filePath) !== '.csv') {
        console.log("Error: File is not in CSV format.");
        return false;
    }
    return true;
}

function parseCSV(filePath) {
    const data = fs.readFileSync(filePath, 'utf8');
    const lines = data.split('\n').filter(line => line.trim() !== '');
    return lines;
}

async function main() {
    const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
    let filePath;
    while (true) {
        filePath = await new Promise(resolve => rl.question("Enter dataset file path: ", resolve));
        if (validateFilePath(filePath)) break;
    }

    try {
        const lines = parseCSV(filePath);
        const totalRecords = lines.length - 1; // ヘッダーを除外
        console.log("=== Record Count ===");
        console.log("Total records:", totalRecords);
    } catch (err) {
        console.log("Error processing file:", err.message);
    } finally {
        rl.close();
    }
}

main();

File: MIDTERM-LAB-3/Java/MP01.java
import java.io.*;
import java.util.*;

public class MP01 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter dataset file path: ");
        String path = input.nextLine();

        File file = new File(path);
        if (!file.exists() || !file.isFile()) {
            System.out.println("Invalid file path.");
            return;
        }

        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
            String header = br.readLine(); // ヘッダー行を読み飛ばす
            int total = 0;
            while (br.readLine() != null) {
                total++;
            }
            System.out.println("=== Record Count ===");
            System.out.println("Total records: " + total);
        } catch (IOException e) {
            System.out.println("Error reading file: " + e.getMessage());
        } finally {
            input.close();
        }
    }
}
